src/Bicep.Core/Features/FeatureProvider.cs

```diff
@@ -3,34 +3,35 @@
 
 using System;
 using System.IO;
-using System.Threading;
+using Bicep.Core.Configuration;
 
 namespace Bicep.Core.Features
 {
     public class FeatureProvider : IFeatureProvider
     {
-        private Lazy<string> cacheRootDirectoryLazy = new(() => GetCacheRootDirectory(Environment.GetEnvironmentVariable("BICEP_CACHE_DIRECTORY")), LazyThreadSafetyMode.PublicationOnly);
-        public string CacheRootDirectory => cacheRootDirectoryLazy.Value;
+        private readonly RootConfiguration configuration;
 
-        public bool RegistryEnabled => true;
+        public FeatureProvider(RootConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
 
-        private Lazy<bool> symbolicNameCodegenEnabledLazy = new(() => ReadBooleanEnvVar("BICEP_SYMBOLIC_NAME_CODEGEN_EXPERIMENTAL", defaultValue: false), LazyThreadSafetyMode.PublicationOnly);
-        public bool SymbolicNameCodegenEnabled => symbolicNameCodegenEnabledLazy.Value;
+        public string CacheRootDirectory => GetCacheRootDirectory(configuration.CacheRootDirectory);
 
-        private Lazy<bool> importsEnabledLazy = new(() => ReadBooleanEnvVar("BICEP_IMPORTS_ENABLED_EXPERIMENTAL", defaultValue: false), LazyThreadSafetyMode.PublicationOnly);
-        public bool ImportsEnabled => importsEnabledLazy.Value;
+        public bool RegistryEnabled => true;
+
```

Comment:
Should we remove this? (feel free to create a follow-up issue)

Reply:
Probably. A fair number of tests explicitly disable the registry, so there may be something to address there.